src/main/scala/firrtl/ExecutionOptionsManager.scala

```diff
@@ -2,584 +2,60 @@
-        customTransforms = firrtlOptions.customTransforms :+ new passes.memlib.InferReadWrite
-      )
-    }.text {
-      "Enable readwrite port inference for the target circuit"
-    }
-
-  parser.opt[String]("repl-seq-mem")
-    .abbr("frsq")
-    .valueName ("-c:<circuit>:-i:<filename>:-o:<filename>")
-    .foreach { x =>
-      firrtlOptions = firrtlOptions.copy(
-        annotations = firrtlOptions.annotations :+ ReplSeqMemAnnotation.parse(x),
-        customTransforms = firrtlOptions.customTransforms :+ new passes.memlib.ReplSeqMem
-      )
-    }
-    .text {
-      "Replace sequential memories with blackboxes + configuration file"
-    }
-
-  parser.opt[String]("list-clocks")
-    .abbr("clks")
-    .valueName ("-c:<circuit>:-m:<module>:-o:<filename>")
-    .foreach { x =>
-      firrtlOptions = firrtlOptions.copy(
-        annotations = firrtlOptions.annotations :+ ClockListAnnotation.parse(x),
-        customTransforms = firrtlOptions.customTransforms :+ new passes.clocklist.ClockListTransform
-      )
-    }
-    .text {
-      "List which signal drives each clock of every descendent of specified module"
-    }
-
-  parser.opt[Unit]("split-modules")
-    .abbr("fsm")
-    .validate { x =>
-      if (firrtlOptions.outputFileNameOverride.nonEmpty)
-        parser.failure("Cannot split-modules if output-file is specified")
-      else parser.success
-    }
-    .foreach { _ =>
-      firrtlOptions = firrtlOptions.copy(emitOneFilePerModule = true)
-    }.text {
-      "Emit each module to its own file in the target directory."
-    }
-
-  parser.opt[Unit]("no-check-comb-loops")
-    .foreach { _ =>
-      firrtlOptions = firrtlOptions.copy(dontCheckCombLoops = true)
-    }.text {
-      "Do NOT check for combinational loops (not recommended)"
-    }
-
-  parser.opt[Unit]("no-dce")
-    .foreach { _ =>
-      firrtlOptions = firrtlOptions.copy(noDCE = true)
-    }.text {
-      "Do NOT run dead code elimination"
-    }
-
-  parser.note("")
-}
-
-sealed trait FirrtlExecutionResult
-
-object FirrtlExecutionSuccess {
-  def apply(
-    emitType    : String,
-    emitted     : String,
-    circuitState: CircuitState
-  ): FirrtlExecutionSuccess = new FirrtlExecutionSuccess(emitType, emitted, circuitState)
-
-
-  def unapply(arg: FirrtlExecutionSuccess): Option[(String, String)] = {
-    Some((arg.emitType, arg.emitted))
+sealed class AugmentedOptionParser(name: String) extends OptionParser[AnnotationSeq](name) {
+  var terminateOnExit = true
+  override def terminate(exitState: Either[String, Unit]): Unit =
+    if (terminateOnExit) sys.exit(0)
+
+  /* Check that no long or short option declarations are duplicated */
+  override def parse(args: Seq[String], init: AnnotationSeq): Option[AnnotationSeq] = {
+    val longDups = options.map(_.name).groupBy(identity).collect{ case (k, v) if v.size > 1 && k != "" => k }
+    val shortDups = options.map(_.shortOpt).flatten.groupBy(identity).collect{ case (k, v) if v.size > 1 => k }
```

Comment:
`.map(...).flatten` => `.flatMap(...)`

Also I didn't know you could do `.groupBy(identity)` cool! I usually do `.groupBy(x => x)` but I like `groupBy(identity)` better